packages/template-retail-react-app/app/contexts/customer-product-lists.js

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React, {useContext} from 'react'
+import {useCurrentCustomer} from '../hooks/use-current-customer'
+import PropTypes from 'prop-types'
+import {CustomerProductListsContext} from './index'
+import {useCustomerProductLists, useShopperCustomersMutation} from 'commerce-sdk-react-preview'
+
+export const CustomerProductListsProvider = ({children}) => {
+    const {data: customer} = useCurrentCustomer()
+    const {isRegistered, customerId} = customer
+    const createCustomerProductList = useShopperCustomersMutation('createCustomerProductList')
+    const {data: productLists, ...restOfQuery} = useCustomerProductLists(
```

Comment:
Just thinking about our template retail react app template...

We are not really leveraging the SCAPI ProductLists feature that allows user to create multiple lists. In our template, we basically just use 1 product list that always has the id `wish_list`... So I'd imagine that we don't need `useCustomerProductLists` in retail react app? I feel like we over complicated the implementation.

What if we remove `CustomerProductListsProvider` and `useProductLists`, just have a single custom hook `useWishList` whose implementation looks like:

```
import {useCustomerProductList} from 'commerce-sdk-react-preview'
const useWishlist = () => {
    const {data: customer} = useCurrentCustomer()
    const createCustomerProductList = useShopperCustomersMutation('createCustomerProductList')
    return useCustomerProductList({
         parameters: {customerId: customer.customerId, listId: 'wish_list'}
    }, {
         // Warning: pseudo code!!!
        onError: (e) => {
            // wishlist doesn't exist, we create it
            if (e.statusCode === 404) {
                createCustomerProductList.mutate({
                        parameters: {customerId},
                        body: {type: 'wish_list'}
                    })
                    // I'm not sure if the query is invalidated by our current cache implementation
                    // gotta test it
            }
        }
    })
}
```

Reply:
Agreed, let's just use put it inside useWishlist for now. We can start thinking about provider when more cases arise